src/test/kotlin/uk/gov/justice/digital/hmpps/pollpush/services/health/HealthCheckIntegrationTest.kt

```diff
@@ -57,6 +67,101 @@ class HealthCheckIntegrationTest : IntegrationTest() {
     assertThat(response.statusCodeValue).isEqualTo(503)
   }
 
+  @Test
+  fun `Queue Health page reports ok`() {
+    subPing(200)
+
+    val response = restTemplate.getForEntity("/health", String::class.java)
+
+    assertThatJson(response.body).node("components.queueHealth.status").isEqualTo("UP")
+    assertThatJson(response.body).node("status").isEqualTo("UP")
+    assertThat(response.statusCodeValue).isEqualTo(200)
+  }
+
+  @Test
+  fun `Queue Health page reports interesting attributes`() {
+    subPing(200)
+
+    val response = restTemplate.getForEntity("/health", String::class.java)
+
+    assertThatJson(response.body).node("components.queueHealth.details.${QueueAttributes.MESSAGES_ON_QUEUE.healthName}").isEqualTo(0)
+    assertThatJson(response.body).node("components.queueHealth.details.${QueueAttributes.MESSAGES_IN_FLIGHT.healthName}").isEqualTo(0)
+  }
+
+  @Test
+  fun `Queue does not exist reports down`() {
+    val realQueueName = ReflectionTestUtils.getField(queueHealth, "queueName") as String
+    ReflectionTestUtils.setField(queueHealth, "queueName", "missing_queue")
+    subPing(200)
+
+    val response = restTemplate.getForEntity("/health", String::class.java)
+
+    assertThatJson(response.body).node("status").isEqualTo("DOWN")
+    assertThatJson(response.body).node("components.queueHealth.status").isEqualTo("DOWN")
+    assertThat(response.statusCodeValue).isEqualTo(503)
+
+    ReflectionTestUtils.setField(queueHealth, "queueName", realQueueName)
```

Comment:
should this go in the `@After` instead otherwise this test failing will cause others to fail too?